esprima.js

```diff
@@ -4095,6 +4100,56 @@
         return list;
     }
 
+
+    function tokenizerState() {
```

Comment:
Admitting that I'm still relatively green when it comes to understanding every PR, it seems fishy that this change would be the first to require a double lookahead and restore.